### PR TITLE
HPCC-20818 User cannot change LDAP password

### PR DESCRIPTION
**Pull request by RussWhitehead**
Attempt to change user password by clicking the username
under "LOGGED IN AS:" results in "Username/password don't match"
This caused because we no longer cache the user's password. This PR
removes the strcmp of the current password, and relies on the authenticate
call to LDAP to ensure the current password is correct

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: system/security/LdapSecurity/ldapsecurity.cpp [hunk 1]
```diff
@@ -1310,12 +1310,7 @@ bool CLdapSecManager::updateUserPassword(ISecUser& user, const char* newPassword
         return false;
     }
 
-    //Update password if authenticated
     bool ok = m_ldap_client->updateUserPassword(user, newPassword, currPassword);
-    if(ok && m_permissionsCache->isCacheEnabled() && !m_usercache_off)
-    {
-        m_permissionsCache->removeFromUserCache(user);
-    }
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapsecurity.cpp]: Do you still cache password? I guess not.

**Review by wangkx — COMMENTED** [review]
just one question

File: system/security/LdapSecurity/ldapsecurity.cpp (hunk 1 above)
Comment (RussWhitehead) [reply system/security/LdapSecurity/ldapsecurity.cpp]: You are correct, we no longer require the password to be stored in the user cache

**Comment by RussWhitehead** [comment]
@wangkx Are you okay with merging this ?

**Pull request merged**